### PR TITLE
Pause playback for topics

### PR DESCRIPTION
**Pull request by tdenewiler**
Added support for pausing when specified topics are about to be published. Can be turned on and off during playback.

It would be a nice feature of rosbag play if you had the ability to pause when specific topics are published. This way you could let a bag run until a particular sensor reading occurs and then see what the immediate effect of that data is on the rest of the system.

### CODE REVIEW

**Comment by tdenewiler** [comment]
I ran unit tests twice, once after checking out `hydro-devel` branch and once after checking out `pause_topics` branch. The steps:
1. `catkin_make`
2. `catkin_make run_tests` so that test bag files are downloaded
3. `catkin_make run_tests_test_rosbag`

This resulted in all of the tests passing except for one (the same test fails in both branches):

```
[ RUN      ] rosbag.read_indexed_1_2_succeeds unknown file: Failure C++ exception with description "Error reading version line" thrown in the test body.
[  FAILED  ] rosbag.read_indexed_1_2_succeeds (0 ms)
```

The changes made in `pause_topics` do not appear to have caused any regressions based on existing unit tests.

**Comment by dirk-thomas** [comment]
Thank you for the PR. It will take some time until we will can review and merge it.

File: tools/rosbag/include/rosbag/player.h [hunk 1]
```diff
@@ -90,6 +90,7 @@ struct ROSBAG_DECL PlayerOptions
 
     std::vector<std::string> bags;
     std::vector<std::string> topics;
+    std::vector<std::string> pausetopics;
```
Comment (dirk-thomas) [inline tools/rosbag/include/rosbag/player.h]: Can you please update this name and all its usage including the command line to something including a separator: either `pause_topics` or `paused_topics`.

**Comment by gerkey** [comment]
Thanks again for the patch.

I'm replacing this PR by #595, targeting Indigo.  We're only putting critical bug fixes into Hydro these days.

**Pull request closed without merging**